src/tools/scripts/test_template_changes.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+######################################
+## Custom Web Almanac script        ##
+######################################
+#
+# This script installs all the required dependencies needed to run the
+# Web Almanac website providing you have python and node installed.
+#
+# It also runs our tests to ensure the website is working for all pages.
+#
+# It is used by various GitHub actions to build and test the site.
+#
+
+# exit when any command fails instead of trying to continue on
+set -e
+
+TEMP_TEMPLATES_DIRECTORY=templates_new
+DIFF_FILENAME=/tmp/template_differences.txt
+
+# This script must be run from src directory
+if [ -d "src" ]; then
+  cd src
+fi
+
+if [ -d "${TEMP_TEMPLATES_DIRECTORY}" ]; then
+  echo "${TEMP_TEMPLATES_DIRECTORY} already exists. Exiting"
+  exit 1
+fi
+
+echo "Installing node modules"
+npm install
+
+echo "Building website"
+npm run generate
+
+echo "Backing up templates"
+cp -r templates "${TEMP_TEMPLATES_DIRECTORY}"
+
+echo "Checkout main branch"
+git checkout main
+
+echo "Reinstalling node modules"
+rm -rf node_modules
+npm install
+
+echo "Building website"
+npm run generate
+
+echo "Diff the two folders"
+#Don't fail if there are differences so turn that check off that temporarily
+set +e
+diff -r templates "${TEMP_TEMPLATES_DIRECTORY}" > "${DIFF_FILENAME}"
+set -e
+
+echo "Differences:"
+cat "${DIFF_FILENAME}"
+
+NUM_DIFFS=$(wc -l < "${DIFF_FILENAME}")
+if [ "${NUM_DIFFS}" -ne "0" ]; then
+  ESCAPED_OUTPUT=$(sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/%0A/g' -e 's/\r/%0D/g' -e 's/\%/%25/g' "${DIFF_FILENAME}")
```

Comment:
In one of my scripts I did this using a custom `escapify` function. https://github.com/ibnesayeed/repo-attrs/blob/26b8384fe72cfecbd0cb56ed99643cfc85a92302/entrypoint.sh#L4-L9

Reply:
Nice. But since for this script it's in a file anyway I think using `sed` is just as good and potentially gives us more powerful options should they be needed.